idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2813 @@
+     ]
+    }
+   ],
+   "source": [
+    "from idaes.core.util import DiagnosticsToolbox\n",
+    "\n",
+    "help(DiagnosticsToolbox)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fe1f5a62",
+   "metadata": {},
+   "source": [
+    "The help() function provides comprehensive information on the DiagnosticsToolbox and all its supported methods. However, it's essential to focus on the initial steps outlined at the beginning of the docstring to get started effectively.\n",
+    "\n",
+    "Here's a breakdown of the steps to start with:\n",
+    "\n",
+    "- `Instantiate Model:` Ensure you have an instance of the model with a degrees of freedom equal to 0.\n",
+    "\n",
+    "- `Create DiagnosticsToolbox Instance:` Next, instantiate a DiagnosticsToolbox object.\n",
+    "\n",
+    "- `Provide Model to DiagnosticsToolbox:` Pass the model instance to the DiagnosticsToolbox.\n",
+    "\n",
+    "- `Call report_structural_issues() Function:` Finally, call the report_structural_issues() function. This function will highlight any warnings in the model's structure, such as unit inconsistencies or other issues related to variables in the caution section.\n",
+    "\n",
+    "By following these steps, you can efficiently utilize the DiagnosticsToolbox to identify and address any structural issues or warnings in your model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "307ec415",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "====================================================================================\n",
+      "Model Statistics\n",
+      "\n",
+      "        Activated Blocks: 21 (Deactivated: 0)\n",
+      "        Free Variables in Activated Constraints: 16 (External: 0)\n",
+      "            Free Variables with only lower bounds: 8\n",
+      "            Free Variables with only upper bounds: 0\n",
+      "            Free Variables with upper and lower bounds: 0\n",
+      "        Fixed Variables in Activated Constraints: 8 (External: 0)\n",
+      "        Activated Equality Constraints: 16 (Deactivated: 0)\n",
+      "        Activated Inequality Constraints: 0 (Deactivated: 0)\n",
+      "        Activated Objectives: 0 (Deactivated: 0)\n",
+      "\n",
+      "------------------------------------------------------------------------------------\n",
+      "0 WARNINGS\n",
+      "\n",
+      "    No warnings found!\n",
+      "\n",
+      "------------------------------------------------------------------------------------\n",
+      "2 Cautions\n",
+      "\n",
+      "    Caution: 3 variables fixed to 0\n",
+      "    Caution: 10 unused variables (4 fixed)\n",
+      "\n",
+      "------------------------------------------------------------------------------------\n",
+      "Suggested next steps:\n",
+      "\n",
+      "    Try to initialize/solve your model and then call report_numerical_issues()\n",
```

Comment:
Similarly, only include solver output from after the diagnostics tips are fixed so that there is only one, clean solver output. The goal of this notebook is to demonstrate creating a custom unit model, not the diagnostics tool itself. If we have an example of using the diagnostics, you can add a link to that notebook.